sherpa-onnx/csrc/provider-config.cc

```diff
@@ -0,0 +1,155 @@
+// sherpa-onnx/csrc/provider-config.cc
+//
+// Copyright (c)  2024  Uniphore (Author: Manickavela)
+
+#include "sherpa-onnx/csrc/provider-config.h"
+
+#include <sstream>
+
+#include "sherpa-onnx/csrc/file-utils.h"
+#include "sherpa-onnx/csrc/macros.h"
+
+namespace sherpa_onnx {
+
+void CudaConfig::Register(ParseOptions *po) {
+  po->Register("cuda-cudnn-conv-algo-search", &cudnn_conv_algo_search,
+          "CuDNN convolution algrorithm search");
+}
+
+bool CudaConfig::Validate() const {
+  if (cudnn_conv_algo_search < 1 && cudnn_conv_algo_search > 3) {
+    SHERPA_ONNX_LOGE("cudnn_conv_algo_search: '%d' is not a valid option."
+                     "Options : [1,3]. Check OnnxRT docs",
+                    cudnn_conv_algo_search);
+    return false;
+  }
+  return true;
+}
+
+std::string CudaConfig::ToString() const {
+  std::ostringstream os;
+
+  os << "CudaConfig(";
+  os << "cudnn_conv_algo_search=\"" << cudnn_conv_algo_search << ")";
+
+  return os.str();
+}
+
+void TensorrtConfig::Register(ParseOptions *po) {
+  po->Register("trt-max-workspace-size", &trt_max_workspace_size,
+              "");
+  po->Register("trt-max-partition-iterations", &trt_max_partition_iterations,
+              "");
+  po->Register("trt-min-subgraph-size ", &trt_min_subgraph_size,
+              "");
+  po->Register("trt-fp16-enable", &trt_fp16_enable,
+              "true to enable fp16");
+  po->Register("trt-detailed-build-log", &trt_detailed_build_log,
+              "true to print TensorRT build logs");
+  po->Register("trt-engine-cache-enable", &trt_engine_cache_enable,
+              "true to enable engine caching");
+  po->Register("trt-engine-cache-path", &trt_engine_cache_path,
+              "");
+  po->Register("trt-timing-cache-enable", &trt_timing_cache_enable,
+              "true to enable timing cache");
+  po->Register("trt-timing-cache-path", &trt_timing_cache_path,
+              "");
+  po->Register("trt-dump-subgraphs", &trt_dump_subgraphs,
+              "true to dump subgraphs");
+}
+
+bool TensorrtConfig::Validate() const {
+  if (trt_max_workspace_size > 0) {
+    SHERPA_ONNX_LOGE("trt_max_workspace_size: '%d' is not valid.",
+        trt_max_workspace_size);
+    return false;
+  }
+  if (trt_max_partition_iterations > 0) {
+    SHERPA_ONNX_LOGE("trt_max_partition_iterations: '%d' is not valid.",
+        trt_max_partition_iterations);
+    return false;
+  }
+  if (trt_min_subgraph_size > 0) {
+    SHERPA_ONNX_LOGE("trt_min_subgraph_size: '%d' is not valid.",
+        trt_min_subgraph_size);
+    return false;
+  }
+  if (trt_fp16_enable != true || trt_fp16_enable != false) {
```

Comment:
What is the point of the test here?
A bool is either true or false. There is not a third possibility.

This test always fails. Have you tested the code locally with tensorrt?

Reply:
I havn't tested the params recently, I will check and get back

Reply:
Apologies, Seemed to missed few commits from local.
Now it is sorted,
But the intention of _if(true||false)_ here is to validate bool or some other integer

Reply:
> What is the point of the test here? A bool is either true or false. There is not a third possibility.
> 
> This test always fails. Have you tested the code locally with tensorrt?

It was always failing because trt_config.Register(po) was missing, I think the error would have been for _not a valida option_ , if that wasn't the case, I will debug further